linker/Mono.Linker/DirectoryAssemblyResolver.cs

```diff
@@ -0,0 +1,118 @@
+using System;
+using System.Reflection;
+using System.Collections.Generic;
+using System.IO;
+using Mono.Collections.Generic;
+using Mono.Cecil;
+
```

Comment:
As an fyi, I made some changes related to the AssemblyResolver in our fork.  We have a custom assembly resolver that we need to use.  Sounds like you have a similar problem.

Here is the PR to our fork that will allow us to rebase on top of the latest linker code.  (I will be upstreaming most of this soon)
https://github.com/Unity-Technologies/linker/pull/1

If you want to see what we did, Search for ILinkerAssemblyResolver.  That's the abstract I came up with to let us hook in our own resolver.